DSharpPlus/Entities/AuditLogParser.cs

```diff
@@ -0,0 +1,1651 @@
+                {
+                    permissionEntry.ApplicationId = (ulong)auditLogAction.TargetId;
+                    permissionEntry.ApplicationCommandId = null;
+                }
+                else
+                {
+                    permissionEntry.ApplicationId = auditLogAction.Options.ApplicationId;
+                    permissionEntry.ApplicationCommandId = auditLogAction.TargetId;
+                }
+
+                foreach (AuditLogActionChange change in auditLogAction.Changes)
+                {
+                    DiscordApplicationCommandPermission? oldValue = ((JObject?)change
+                            .OldValue)?
+                        .ToDiscordObject<DiscordApplicationCommandPermission>();
+
+                    DiscordApplicationCommandPermission? newValue = ((JObject)change
+                            .NewValue)?
+                        .ToDiscordObject<DiscordApplicationCommandPermission>();
+
+                    permissionEntry.PermissionChanges.Append(new PropertyChange<DiscordApplicationCommandPermission>
+                    {
+                        Before = oldValue != null ? oldValue : null, After = newValue != null ? newValue : null
+                    });
+                }
+
+                break;
+
+            case AuditLogActionType.AutoModerationBlockMessage:
+            case AuditLogActionType.AutoModerationFlagToChannel:
+            case AuditLogActionType.AutoModerationUserCommunicationDisabled:
+                entry = new DiscordAuditLogAutoModerationExecutedEntry();
+
+                DiscordAuditLogAutoModerationExecutedEntry autoModerationEntry =
+                    entry as DiscordAuditLogAutoModerationExecutedEntry;
+
+                autoModerationEntry.TargetUser =
+                    members.TryGetValue(auditLogAction.TargetId.Value, out DiscordMember? targetMember)
+                        ? targetMember
+                        : new DiscordUser {Id = auditLogAction.TargetId.Value, Discord = guild.Discord};
+
+                autoModerationEntry.ResponsibleRule = auditLogAction.Options.RoleName;
+                autoModerationEntry.Channel = guild.GetChannel(auditLogAction.Options.ChannelId);
+                autoModerationEntry.RuleTriggerType =
+                    (RuleTriggerType)int.Parse(auditLogAction.Options.AutoModerationRuleTriggerType);
+                break;
+
+            case AuditLogActionType.AutoModerationRuleCreate:
+            case AuditLogActionType.AutoModerationRuleUpdate:
+            case AuditLogActionType.AutoModerationRuleDelete:
+                entry = ParseAutoModerationRuleUpdateEntry(guild, auditLogAction);
+                break;
+
+            default:
+                guild.Discord.Logger.LogWarning(LoggerEvents.AuditLog,
+                    "Unknown audit log action type: {0} - this should be reported to library developers",
+                    (int)auditLogAction.ActionType);
+                break;
```

Comment:
this shouldn't be a warning by default - or at least be toggleable, similar to the unknown events warning

Reply:
I think i will make it toggleable

Reply:
probably a good idea, especially considering we know there's undocumented types